coderd/metricscache/metricscache.go

```diff
@@ -214,7 +216,9 @@ func (c *Cache) refreshTemplateDAUs(ctx context.Context) error {
 		return xerrors.Errorf("deployment daus: %w", err)
 	}
 
-	for _, template := range templates {
+	ids := make([]uuid.UUID, len(templates))
+	for i, template := range templates {
+		ids[i] = template.ID
```

Comment:
We've been doing this to prevent slices from having nils/empty values. You still get the benefit of pre-allocating. 
```suggestion
	ids := make([]uuid.UUID, 0, len(templates))
	for _, template := range templates {
		ids = append(ids, template.ID)
```

Reply:
ah a good cap v len in the wild, great catch